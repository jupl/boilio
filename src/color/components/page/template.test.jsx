import React from 'react'
import renderer from 'react-test-renderer'
import {shallow} from 'enzyme'
import ColorPage, {Button} from './template'

describe('<Color.ColorPage> Template', () => {
  const props = {
    color: 'white',
    onPreviousColor: jest.fn(),
    onNextColor: jest.fn(),
  }

  it('should render as expected', () => {
    const component = renderer.create(<ColorPage {...props} />)
    expect(component.toJSON()).toMatchSnapshot()
  })

  it('should invoke events as expected', () => {
    const component = shallow(<ColorPage {...props} />)
    const buttons = component.find(Button)
    const previousButton = buttons.at(0)
    const nextButton = buttons.at(1)

    expect(props.onPreviousColor).not.toBeCalled()
    expect(props.onNextColor).not.toBeCalled()

    previousButton.simulate('click')
    expect(props.onPreviousColor.mock.calls.length).toBe(1)
    expect(props.onNextColor).not.toHaveBeenCalled()

    nextButton.simulate('click')
    expect(props.onPreviousColor.mock.calls.length).toBe(1)
    expect(props.onNextColor.mock.calls.length).toBe(1)
  })
})
