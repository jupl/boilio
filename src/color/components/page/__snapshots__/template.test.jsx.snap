exports[`<Color.ColorPage> Template should render as expected 1`] = `
<div
  className="cPPbgF"
  color="white">
  <div
    className="ijXwvP">
    <button
      className="bwNSRs"
      onClick={[Function]}>
      <
    </button>
    <span
      className="gsqXms">
      Hello, World
    </span>
    <button
      className="bwNSRs"
      onClick={[Function]}>
      >
    </button>
  </div>
</div>
`;
